package/MDAnalysis/core/universe.py

```diff
@@ -337,16 +359,18 @@ def __init__(self, topology=None, *coordinates, all_coordinates=False,
             'all_coordinates': all_coordinates
         }
         self._kwargs.update(kwargs)
-
-        format, topology_format = _resolve_formats(*coordinates, format=format,
-                                                   topology_format=topology_format)
-
-        if not isinstance(topology, Topology) and not topology is None:
+        format, topology_format = _resolve_formats(
+            *coordinates, format=format, topology_format=topology_format)
+        self._begin_guess = False
+        self.txyx = False
+        if not isinstance(topology, Topology) and topology is not None:
             self.filename = _check_file_like(topology)
-            topology = _topology_from_file_like(self.filename,
-                                                topology_format=topology_format,
-                                                **kwargs)
-
+            top = _topology_from_file_like(self.filename,
+                                           topology_format=topology_format,
+                                           **kwargs)
+            topology = top[0]
+            self._begin_guess = top[1]
+            self.txyz = top[2]
```

Comment:
You can write this as

```python
topology, self._begin_guess, self.txyz = _topology_from_file_like(self.filename,
                                           topology_format=topology_format,
                                           **kwargs)
```

In general, every time you ask a tuple, a list, or any collection of a known number of items, you can unpack it this way.

Reply:
I'm not quite clear on why these variables need to be set -- they look like they're only used in this method?